File: easy/max_ice_cream.go
package easy

import "slices"

func MaxIceCream(costs []int, coins int) int {
	cones := 0
	slices.Sort(costs)
	for _, cost := range costs {
		if cost <= coins {
			coins -= cost
			cones++
		}
	}
	return cones
}

File: src/easy/max_ice_cream.rs
pub fn max_ice_cream(costs: Vec<i32>, coins: i32) -> i32 {
    let mut cones: i32 = 0;
    let mut remaining = coins;
    let mut arr = costs;
    arr.sort();
    for &n in arr.iter() {
        if remaining >= n {
            remaining -= n;
            cones += 1;
        }
    }
    cones
}
